client-sdk/ts-web/rt/playground/src/subcall.js

```diff
@@ -0,0 +1,142 @@
+    const sapphireTestnetRuntimeID = oasis.misc.fromHex(
+        '000000000000000000000000000000000000000000000000a6d1e3ebf60dff6c',
+    );
+    const testnetNic = new oasis.client.NodeInternal('https://testnet.grpc.oasis.io');
+
+    const roflmarket = new oasisRT.roflmarket.Wrapper(sapphireTestnetRuntimeID);
+    const rofl = new oasisRT.rofl.Wrapper(sapphireTestnetRuntimeID);
+    const core = new oasisRT.core.Wrapper(sapphireTestnetRuntimeID);
+
+    console.log(
+        'queryCallDataPublicKey testnetNic',
+        await core.queryCallDataPublicKey().query(testnetNic),
+    );
+
+    if (!window.ethereum) throw new Error('No MetaMask installed');
+
+    const [account] = await window.ethereum.request({
+        method: 'eth_requestAccounts',
+    });
+    const client = createWalletClient({
+        account: account,
+        chain: sapphireTestnet,
+        transport: custom(window.ethereum),
+    });
+
+    let hash;
+
+    const createAppTx = rofl
+        .callCreate()
+        .setBody({
+            scheme: oasisRT.types.IdentifierScheme.CreatorNonce,
+            policy: {
+                quotes: {
+                    pcs: {
+                        tcb_validity_period: 30,
+                        min_tcb_evaluation_data_number: 18,
+                        tdx: {},
+                    },
+                },
+                enclaves: [],
+                endorsements: [
+                    {
+                        any: {},
+                    },
+                ],
+                fees: oasisRT.types.FeePolicy.EndorsingNodePays,
+                max_expiration: 3,
+            },
+            metadata: {
+                'net.oasis.rofl.license': 'Apache-2.0',
+                'net.oasis.rofl.name': 'create through subcall',
+                'net.oasis.rofl.repository':
+                    'https://github.com/oasisprotocol/oasis-sdk/tree/main/client-sdk/ts-web',
+                'net.oasis.rofl.version': '0.1.0',
+            },
+        })
+        .toSubcall();
+    hash = await client.sendTransaction(createAppTx);
+    console.log('create app: tx hash', hash);
+    const appId = await waitForAppId(hash);
+    console.log('appId', appId);
+
+    let app = await rofl
+        .queryApp()
+        .setArgs({id: oasisRT.rofl.fromBech32(appId)})
+        .query(testnetNic);
+    console.log('App', app);
+
+    hash = await client.sendTransaction(
+        rofl
+            .callUpdate()
+            .setBody({
+                id: app.id,
+                admin: app.admin,
+                policy: app.policy,
+                metadata: {
+                    ...app.metadata,
+                    'net.oasis.rofl.name': 'create through subcall updated',
+                },
+                secrets: {
+                    ...app.secrets,
+                    a: oasis.misc.fromBase64(
+                        oasisRT.rofl.encryptSecret('a', oasis.misc.fromString('secret'), app.sek),
```

Comment:
TODO: test if rofl app can read the encrypted secrets. It seems that even incorrectly encrypted secrets can be added this way

Reply:
Manually tested with a combination of this + cli. It works. https://explorer.oasis.io/testnet/sapphire/tx/cd562a08c57c636bfb5461365679560c07d8d7cffad2c0e982cf9450157862c7